File: test/src/commonTest/kotlin/Pass.kt
import natives.test.consume
import natives.test.consumeBoolean
import natives.test.consumeByte
import natives.test.consumeChar
import natives.test.consumeDouble
import natives.test.consumeFloat
import natives.test.consumeInt
import natives.test.consumeLong
import natives.test.consumeString
import kotlin.test.Test
import kotlin.test.assertTrue

class Pass {

    @Test
    fun passEmpty() = withLib {
        assertTrue(consume())
    }

    @Test
    fun passInt() = withLib {
        assertTrue(consumeInt(99))
    }

    @Test
    fun passLong() = withLib {
        assertTrue(consumeLong(9223372036854775805L))
    }

    @Test
    fun passFloat() = withLib {
        assertTrue(consumeFloat(99f))
    }

    @Test
    fun passDouble() = withLib {
        assertTrue(consumeDouble(1.0))
    }

    @Test
    fun passByte() = withLib {
        assertTrue(consumeByte(1.toByte()))
    }

    @Test
    fun passBoolean() = withLib {
        assertTrue(consumeBoolean(true))
    }

    @Test
    fun passChar() = withLib {
        assertTrue(consumeChar('a'))
    }

    @Test
    fun passString() = withLib {
        assertTrue(consumeString("test string"))
    }
}

File: tests/natives/test/src/pass.c
#include <api.h>
#include <string.h>

// Consume

KBoolean consume() {
    return true;
}

KBoolean consumeInt(KInt arg) {
    return arg == 99;
}

KBoolean consumeLong(KLong arg) {
    return arg == 9223372036854775805;
}

KBoolean consumeFloat(KFloat arg) {
    return arg == 99.9f;
}

KBoolean consumeDouble(KDouble arg) {
    return arg == 1.1;
}

KBoolean consumeByte(KByte arg) {
    return arg == 1;
}

KBoolean consumeBoolean(KBoolean arg) {
    return arg == true;
}

KBoolean consumeChar(KChar arg) {
    return arg == 'a';
}

KBoolean consumeString(KString arg) {
    return arg.length == 11 && strncmp(arg.data, "test string", arg.length) == 0;
}